src/WebJobs.Script/Workers/Rpc/RpcWorkerChannel.cs

```diff
@@ -111,7 +111,7 @@ internal class RpcWorkerChannel : IRpcWorkerChannel, IDisposable
 
         public bool IsChannelReadyForInvocations()
         {
-            return _state.HasFlag(RpcWorkerChannelState.InvocationBuffersInitialized | RpcWorkerChannelState.Initialized);
+            return !_disposing && !_disposed && _state.HasFlag(RpcWorkerChannelState.InvocationBuffersInitialized | RpcWorkerChannelState.Initialized);
```

Comment:
Add test where this method returns false if a restart is issued on the channel

Reply:
Added a test where this returns false when it's being disposed.